apps/etterlatte-vilkaarsvurdering-kafka/src/main/kotlin/vilkaarsvurdering/services/VilkaarsvurderingService.kt

```diff
@@ -0,0 +1,26 @@
+package no.nav.etterlatte.vilkaarsvurdering.services
+
+import io.ktor.client.HttpClient
+import io.ktor.client.request.post
+import io.ktor.client.request.setBody
+import io.ktor.http.ContentType
+import io.ktor.http.contentType
+import kotlinx.coroutines.runBlocking
+import java.util.*
+
+interface VilkaarsvurderingService {
+    fun kopierForrigeVilkaarsvurdering(behandlingId: UUID, behandlingViOmregnerFra: UUID)
+}
+
+class VilkaarsvurderingServiceImpl(private val VilkaarsvurderingKlient: HttpClient, private val url: String) :
+    VilkaarsvurderingService {
+    override fun kopierForrigeVilkaarsvurdering(behandlingId: UUID, behandlingViOmregnerFra: UUID) {
+        runBlocking {
+            VilkaarsvurderingKlient.post("$url/api/$behandlingId/kopier") {
```

Comment:
Kunne det vært en ide at man her heller oppretter en vilkårsvurdering som man ellers ville ha gjort, men at denne da i servicen kopieres etc når servicen i vilkårsvurderingsappen ser at dette er en omregning? Da slipper konsumenten og vite den type implementasjonsdetaljer. Kan bare spørre om å få en vilkårsvurdering og får da en som er ferdig godkjent og alt er klart for beregning?

Reply:
Ja det er en mulighet det og. Det er et godt poeng at konsument slipper å tenke på implementasjonsdetaljer, men føler det blir litt det samme bare tvertom, at nå produsenten må vite hvordan sine konsumenter ser ut eller hvordan de fungerer. Begge ur uheldige uansett 😛

Jeg tenker at jeg i første omgang bare får in dette og får testet det så vi har noe som fungerer, og så kan vi vurdere om vi ønsker å gjøre om det (skal i utgangspunkt ikke være så krevende). ☺️